stable/datadog/templates/container-agent.yaml

```diff
@@ -0,0 +1,161 @@
+    {{- end }}
+    {{- if .Values.datadog.collectEvents }}
+    - name: DD_COLLECT_KUBERNETES_EVENTS
+      value: {{.Values.datadog.collectEvents | quote}}
+    {{- end }}
+    {{- else }}
+    - name: DD_CLUSTER_AGENT_ENABLED
+      value: {{ .Values.clusterAgent.enabled | quote }}
+    - name: DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME
+      value: {{ template "datadog.fullname" . }}-cluster-agent
+    - name: DD_CLUSTER_AGENT_AUTH_TOKEN
+      valueFrom:
+        secretKeyRef:
+          name: {{ template "datadog.fullname" . }}-cluster-agent
+          key: token
+    {{- end }}
+    {{- if .Values.datadog.podLabelsAsTags }}
+    - name: DD_KUBERNETES_POD_LABELS_AS_TAGS
+      value: '{{ toJson .Values.datadog.podLabelsAsTags }}'
+    {{- end }}
+    {{- if .Values.datadog.podAnnotationsAsTags }}
+    - name: DD_KUBERNETES_POD_ANNOTATIONS_AS_TAGS
+      value: '{{ toJson .Values.datadog.podAnnotationsAsTags }}'
+    {{- end }}
+    {{- if semverCompare "^1.7-0" .Capabilities.KubeVersion.GitVersion }}
+    - name: DD_KUBERNETES_KUBELET_HOST
+      valueFrom:
+        fieldRef:
+          fieldPath: status.hostIP
+    {{- end }}
+    {{- if .Values.datadog.logsEnabled }}
+    - name: DD_LOGS_ENABLED
+      value: {{.Values.datadog.logsEnabled | quote}}
+    {{- end }}
+    {{- if .Values.datadog.logsConfigContainerCollectAll }}
+    - name: DD_LOGS_CONFIG_CONTAINER_COLLECT_ALL
+      value: {{.Values.datadog.logsConfigContainerCollectAll | quote}}
+    {{- end }}
+    {{- if .Values.datadog.criSocketPath }}
+    - name: DD_CRI_SOCKET_PATH
+      value: {{ .Values.datadog.criSocketPath | quote }}
+    {{- end }}
+    {{- if not .Values.datadog.livenessProbe }}
+    - name: DD_HEALTH_PORT
+      value: "5555"
+    {{- end }}
+    {{- if .Values.datadog.useDogStatsDSocketVolume }}
+    - name: DD_DOGSTATSD_SOCKET
+      value: "/var/run/datadog/dsd.socket"
+    {{- end }}
+    {{- if and .Values.clusterAgent.clusterChecks.enabled  (not .Values.clusterchecksDeployment.enabled) }}
+    - name: DD_EXTRA_CONFIG_PROVIDERS
+      value: "clusterchecks"
+    {{- end }}
+{{- if .Values.datadog.env }}
+{{ toYaml .Values.datadog.env | indent 4 }}
+{{- end }}
+  volumeMounts:
+    - name: config
+      mountPath: /etc/datadog-agent
+    {{- if .Values.datadog.useCriSocketVolume }}
+    - name: runtimesocket
+      mountPath: {{ default "/var/run/docker.sock" .Values.datadog.criSocketPath | quote }}
+      readOnly: true
+    {{- end }}
+    {{- if .Values.datadog.useDogStatsDSocketVolume }}
+    - name: dsdsocket
+      mountPath: "/var/run/datadog"
+    {{- end }}
+    - name: procdir
+      mountPath: /host/proc
+      readOnly: true
+    - name: cgroups
+      mountPath: /host/sys/fs/cgroup
+      readOnly: true
+    {{- if .Values.datadog.logsEnabled }}
+    - name: pointerdir
+      mountPath: /opt/datadog-agent/run
+    {{- end }}
+{{- if .Values.datadog.volumeMounts }}
+{{ toYaml .Values.datadog.volumeMounts | indent 10 }}
```

Comment:
same issue here

Reply:
These are not shared between containers, only used for the container agent so far. You meant `.Values.datadog.volumes` maybe?